CMakeLists.txt

```diff
@@ -57,7 +57,10 @@ option(ENABLE_PROGRAMS "Build mbed TLS programs." ON)
 option(UNSAFE_BUILD "Allow unsafe builds. These builds ARE NOT SECURE." OFF)
 option(MBEDTLS_FATAL_WARNINGS "Compiler warnings treated as errors" ON)
```

Comment:
Actively modifying a toggle option, using a hard to maintain sed inline replace string can be hard to maintain. 

I would separate the variable declaration and setting like that:

~~~~~~~
option(GEN_FILES "Generate the auto-generated files as needed" OFF)
if(CMAKE_HOST_WIN32)
    # N.B. The comment on the next line is significant! If you change it,
    # edit the sed command in prepare_release.sh that modifies
    # CMakeLists.txt.
    set(GEN_FILES OFF)
else()
    set(GEN_FILES ON)
endif()
~~~~~~~

Then replacing `set(GEN_FILES OFF)` -> `set(GEN_FILES ON)` can be easier. The only drawback to this is if we have checks that test if GEN_FILES  instead of its value (On=1, Off = 0) but I do not believe that is currently the the case. 

I won't block the PR for this minor issue though, so feel free to consider if only if you are making changes.

Reply:
> Actively modifying a toggle option, using a hard to maintain sed inline replace string can be hard to maintain.

I agree, but my preferred resolution would be to drop support for “unrelease” mode, which would remove most of the complexity around the sed expression.

Since neither you nor Tom consider this a blocker, I propose to merge this pull request and follow up later.